### PR TITLE
Implement mimetype categorization for easier file-type / preview detection 

### LINKED ISSUE #10685: Don't expose image binary type
In the reviewers browse api we're exposing the field `binary` which can be a boolean or string but really, should only be a boolean.

We are already exposing the mimetype so the client should be able to handle that properly.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This implements more advanced file-type detection based on `python-magic` thus we don't rely on the files extensions anymore but actually verify their content (header) before serving it to the client and reviewer.

Fixes #10685  (and actually rewrites it's meaning)

### CODE REVIEW

File: docs/topics/api/reviewers.rst [hunk 1]
```diff
@@ -131,7 +131,7 @@ This endpoint allows you to browse through the contents of an Add-on version.
     :>json string file.content: Raw content of the requested file.
     :>json string file.selected_file: The selected file, either from the ``file`` parameter or the default (manifest.json, install.rdf or package.json for Add-ons as well as the XML file for search engines).
     :>json array file.entries[]: The complete file-tree of the extracted XPI.
-    :>json boolean|string file.entries[].binary: ``True`` if the file is a binary file (e.g an .exe, dll, java, swf file), ``'image'`` if the file is an image or ``False`` otherwise. If ``False`` or ``'image'`` the file should be presentable to the user.
+    :>json boolean file.entries[].binary: ``True`` if the file is a binary file (e.g an .exe, dll, java, swf file) or ``False`` otherwise. If ``False`` the file should be presentable to the user.
```
Comment (diox) [inline docs/topics/api/reviewers.rst]: Didn't notice that before: should we say `class` and not `java` ? `java` files are text, technically :)

File: src/olympia/reviewers/serializers.py [hunk 2]
```diff
@@ -123,27 +134,28 @@ def _fetch_entries():
...
-        if ext in denied_extensions:
-            return True
+        if entry.type == 'tree':
+            return 'application/octet-stream', 'directory'
 
-        bytes_ = tuple(bytearray(memoryview(blob)[:4]))
+        # Hardcoding the maximum amount of bytes to read here
+        # until https://github.com/ahupp/python-magic/commit/50e8c856
+        # lands in a release and we can read that value from libmagic
+        bytes_ = io.BytesIO(memoryview(blob)).read(1048576)
```
Comment (diox) [inline src/olympia/reviewers/serializers.py]: I don't get this. Isn't that value the default ? Is something else altering the max value if we don't set it ?

Comment (EnTeQuAk) [reply src/olympia/reviewers/serializers.py]: Updated the comment, the value is the default, correct, but I have to wrap this to not exhaust/read the whole blob into memory again (e.g for large files).

File: docs/topics/api/reviewers.rst (hunk 1 above)
Comment (EnTeQuAk) [reply docs/topics/api/reviewers.rst]: eh, indeed

Comment (EnTeQuAk) [reply docs/topics/api/reviewers.rst]: This needs updating anyway

Comment (EnTeQuAk) [reply docs/topics/api/reviewers.rst]: Updated in 2c0d38b690

File: Dockerfile [hunk 3]
```diff
@@ -28,6 +28,7 @@ RUN apt-get update && apt-get install -y \
         zlib1g-dev \
         libffi-dev \
         libssl-dev \
+        libmagic-dev \
```
Comment (diox) [inline Dockerfile]: You don't need this in `.travis.yml` as well ?

Comment (EnTeQuAk) [reply Dockerfile]: It's already installed in travis.

**Pull request merged**